datacube/index/postgres/_datasets.py

```diff
@@ -116,9 +116,11 @@ def get_derived(self, id_):
         """
         Get all derived datasets
 
-        :param UUID id_: dataset id
+        :param Union[str,UUID] id_: dataset id
         :rtype: list[Dataset]
         """
+        if not isinstance(id_, UUID):
+            id_ = UUID(id_)
```

Comment:
Was this failing? I'm pretty sure the query works with both types. The other nearby methods like `has()` don't seem to convert their input UUID.

Test of the old code on both types:
```
In [3]: len(dc.index.datasets.get_derived("b069121e-8d32-48ed-b5eb-6335d2884bc9"))
Out[3]: 11

In [4]: from uuid import UUID

In [5]: len(dc.index.datasets.get_derived(UUID("b069121e-8d32-48ed-b5eb-6335d2884bc9")))
Out[5]: 11

In [6]: 
```
(... unfortunately it doesn't look like we have tests on those methods in the actual repo, at least not directly. But there are some integration tests in wider repos.)

Reply:
Not failing per se, but the new code will catch the case where a string is passed in that is not a valid UUID.

Reply:
Ah, good point!